lib/mj-single.js

```diff
@@ -315,6 +316,28 @@ function ConfigureMathJax() {
     }
   };
 
+  if (extensions) {
+    //
+    // Parse added extensions list and add to standard ones
+    //
+    var extensionList = extensions.split(/s*,\s*/);
+    for (var i = 0; i < extensionList.length; i++) {
+      var matches = extensionList[i].match(/^(?:(.+)\/)?(.*)$/);
+      var category = matches[1], name = matches[2];
+      if (!category) {
+        window.MathJax.extensions = window.MathJax.extensions.concat(name + '.js');
+      } else {
+        if (!window.MathJax[category]) {
+          window.MathJax[category] = {};
+        }
+        if (!window.MathJax[category].extensions) {
+          window.MathJax[category].extensions = [];
+        }
+        window.MathJax[category].extensions =
+          window.MathJax[category].extensions.concat(name + '.js');
+      }
+    }
+  }
```

Comment:
You co to a lot of effort here to break up the extensions into their individual categories, but that is not strictly necessary.  The main `extensions` array can include things like `"TeX/noUndefined.js"`, so it is OK to leave them all there.  The only reason to move them to the categories is that they would only be loaded when the input jax was loaded, but MathJax-node loads all three input jax automatically, so there is really no savings for that.  Might as well not introduce the added complexity of breaking up the list.